packages/headless/src/controllers/core/smart-snippet-questions-list/headless-core-smart-snippet-questions-list.ts

```diff
@@ -0,0 +1,266 @@
+   */
+  selectSource(identifier: string): void;
+  /**
+   * Prepares to select the source after a certain delay, sending analytics if it hadn't been selected before.
+   *
+   * In a DOM context, we recommend calling this method on the `touchstart` event.
+   *
+   * @param identifier - The `questionAnswerId` of the smart snippet to collapse.
+   */
+  beginDelayedSelectSource(identifier: string): void;
+  /**
+   * Cancels the pending selection caused by `beginDelayedSelect`.
+   *
+   * In a DOM context, we recommend calling this method on the `touchend` event.
+   *
+   * @param identifier - The `questionAnswerId` of the smart snippet to collapse.
+   */
+  cancelPendingSelectSource(identifier: string): void;
+  /**
+   * Selects a link inside an answer, logging a UA event to the Coveo Platform if it was never selected before.
+   *
+   * In a DOM context, we recommend calling this method on all of the following events:
+   * * `contextmenu`
+   * * `click`
+   * * `mouseup`
+   * * `mousedown`
+   *
+   * @param identifier - The `questionAnswerId` of the smart snippet containing the link.
+   * @param link - The link to select.
+   */
+}
+
+/**
+ * The `SmartSnippetQuestionsList` controller allows to manage additional queries for which a SmartSnippet model can provide relevant excerpts.
+ */
+export interface SmartSnippetQuestionsList
+  extends SmartSnippetQuestionsListCore {
+  /**
+   * Selects a link inside an answer, logging a UA event to the Coveo Platform if it was never selected before.
+   *
+   * In a DOM context, we recommend calling this method on all of the following events:
+   * * `contextmenu`
+   * * `click`
+   * * `mouseup`
+   * * `mousedown`
+   *
+   * @param identifier - The `questionAnswerId` of the smart snippet containing the link.
+   * @param link - The link to select.
+   */
+  selectInlineLink(identifier: string, link: InlineLink): void;
+  /**
+   * Prepares to select a link inside an answer after a certain delay, sending analytics if it was never selected before.
+   *
+   * In a DOM context, we recommend calling this method on the `touchstart` event.
+   *
+   * @param identifier - The `questionAnswerId` of the smart snippet containing the link.
+   * @param link - The link to select.
+   */
+  beginDelayedSelectInlineLink(identifier: string, link: InlineLink): void;
+  /**
+   * Cancels the pending selection caused by `beginDelayedSelectInlineLink`.
+   *
+   * In a DOM context, we recommend calling this method on the `touchend` event.
+   *
+   * @param identifier - The `questionAnswerId` of the smart snippet containing the link.
+   * @param link - The link to select.
+   */
+  cancelPendingSelectInlineLink(identifier: string, link: InlineLink): void;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `SmartSnippetQuestionsList` controller.
+ */
+export interface SmartSnippetQuestionsListState {
```

Comment:
I would prefer if this interface was renamed to `SmartSnippetQuestionsListStateCore` similar to the controller's main interface, and each use-case would re-export their own state by extending the core interface.

I think this would help further divide core from search/insight/etc controllers.

Reply:
updated!